src/CustomMarshalers/src/System/Runtime/InteropServices/CustomMarshalers/EnumerableViewOfDispatch.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+
+namespace System.Runtime.InteropServices.CustomMarshalers
+{
+    [Serializable]
+    internal class EnumerableViewOfDispatch : ICustomAdapter, IEnumerable
+    {
+        private const int DispId_NewEnum = -4;
+        private readonly object dispatch;
+
+        public EnumerableViewOfDispatch(object dispatch)
```

Comment:
Does this need to check for null?

Reply:
It does not. The null case is covered in the `EnumerableToDispatchMarshaler.MarshalNativeToManaged` method.